scripts/js/commands/checkPagesRender.ts

```diff
@@ -188,5 +209,13 @@ async function determineFilePaths(args: Arguments): Promise<string[]> {
     globs.push(`${prefix}${glob}`);
   }
   const result = await globby(globs);
-  return result.filter((fp) => !ALLOWLIST.has(fp));
+  return result.filter((fp) => filterPlatformSpecificPage(fp, args.legacy));
+}
+
+function filterPlatformSpecificPage(page: string, legacy?: boolean) {
+  if (legacy) {
+    return !CLOUD_ONLY_PAGES.has(page) && !page.startsWith("/docs/api");
```

Comment:
I think the `/` prefix will mess things up with `/docs/api`?

```suggestion
    // API docs should never be checked with the legacy app because they render
    // identically in the cloud app.
    return !CLOUD_ONLY_PAGES.has(page) && !page.startsWith("docs/api");
```